DOC: `sampling_tdr.rst` failing in CircleCI smoke-tutorials job

Failure from [this log](https://app.circleci.com/pipelines/github/scipy/scipy/28971/workflows/12e90e53-9f1d-4911-894d-4fe6c665204a/jobs/97711):

```
doc/source/tutorial/stats/sampling_srou.rst x                            [ 96%]
doc/source/tutorial/stats/sampling_tdr.rst F                             [100%]

=================================== FAILURES ===================================
__________________________ [doctest] sampling_tdr.rst __________________________
154     527
155     >>> dist2 = StandardNormal()
156     >>> # use the same stream of uniform random numbers
157     >>> rng2 = TransformedDensityRejection(dist2, max_squeeze_hat_ratio=0.999,
158     ...                                    random_state=urng2)
159     >>> dist2.callbacks  # evaluations during setup
160     467
161     >>> dist2.callbacks = 0  # don't consider evaluations during setup
162     >>> rvs = rng2.rvs(100000)
163     >>> dist2.callbacks  # evaluations during sampling
Expected:
    84
Got:
    57

/home/circleci/repo/doc/source/tutorial/stats/sampling_tdr.rst:163: DocTestFailure
=========================== short test summary info ============================
FAILED doc/source/tutorial/stats/sampling_tdr.rst::sampling_tdr.rst
=================== 1 failed, 25 passed, 6 xfailed in 28.12s ===================
```

This tutorial was the first one to be converted to MyST-md (although it is 7 weeks ago, in gh-20303), so perhaps the seeding for the test machinery doesn't work anymore? @melissawm you may be able to say more here?

--- a/doc/source/tutorial/stats/sampling_tdr.rst
+++ b/doc/source/tutorial/stats/sampling_tdr.rst
@@ -161,7 +161,7 @@ distribution:
     >>> dist2.callbacks = 0  # don't consider evaluations during setup
     >>> rvs = rng2.rvs(100000)
     >>> dist2.callbacks  # evaluations during sampling
-    84
+    84      # may vary
 
 As we can see, far fewer PDF evaluations are required during sampling when
 we increase the ``squeeze_hat_ratio``. The PPF-hat function is also more accurate:

